src/utils/storageFacade.utils.ts

```diff
@@ -1,122 +1,355 @@
+  let { data, error, status } = await supabase
+    .from('packages')
+    .select('*')
+    .eq('sessionid', sessionId)
+    .eq('package_ref', pkg.ref);
+
+  if (error) {
+    logger.error(error.message);
+  } else if (data) {
+    if (data.length == 0) {
+      //INSERT
+      return insertPackage(pkg, sessionId);
+    } else {
+      //UPDATE
+      return updatePackage(pkg, sessionId);
+    }
+  }
+  return status;
+}
+
+/**
+ * Read all vulnerabilities in one session
+ * @param sessionId associated with one user
+ * @returns list of vulnerabilities, empty list if none matching present in DB
+ */
+export async function readVulnsBySession(sessionId: number) {
+  let { data, error } = await supabase
+    .from('vulnerabilities')
+    .select('*,junction!inner(packageid,packages!inner(package_ref))')
+    .eq('junction.packages.sessionid', sessionId);
+
+  let cves: Vulnerability[] = [];
+  if (error) {
+    logger.error(error.message);
+  } else if (data) {
+    for (let v of data) {
+      cves.push(
+        //map database result to Vulnerability object
+        {
+          cveId: v.cveidstring,
+          packageRef: v.junction[0].packages.package_ref,
+          impact: v.impact,
+          likelihood: v.likelihood,
+          risk: v.risk,
+          cvss2: v.cvss_vector,
+        }
+      );
+    }
   }
-//Task 7: Double-check if read function overwrites current entry vs using update (applicable for packages & vulnerabilities)
-  //Task #5: Ensure sessionID is not one of the function parameters - need more clarity on sessionid usage before implementation
-/****************** WRITE PACKAGE**********************/
+
+  return cves;
+}
+
+/**
+ * Read all vulnerabilites in one package
+ * @param packageRef unique string ID retrieved from user-uploaded software bill of materials
+ * @param sessionId associated with one user
+ * @returns list of packages, empty list if none matching are present in DB
+ */
+export async function readVulnsByPkg(packageRef: string, sessionId: number) {
+  const { data, error } = await supabase
+    .from('vulnerabilities')
+    .select('*,junction!inner(packageid,packages!inner(package_ref))')
+    .eq('junction.packages.package_ref', packageRef);
+  let cves: Vulnerability[] = [];
+
+  if (error) {
+    logger.error(error.message);
+  } else if (data) {
+    for (let v of data) {
+      cves.push(
+        //map database result to Vulnerability object
+        {
+          cveId: v.cveidstring,
+          packageRef: v.junction[0].packages.package_ref,
+          impact: v.impact,
+          likelihood: v.likelihood,
+          risk: v.risk,
+          cvss2: v.cvss_vector,
+        }
+      );
+    }
+  }
+  return cves;
+}
+
+/**
+ * Write Package, updates if entry exists in database, inserts otherwise
+ * @param cve to write
+ * @param sessionId associated with one user
+ * @returns supabase write status
+ */
+export async function writeVuln(cve: Vulnerability, sessionId: number) {
```

Comment:
function is not writting vulnerabilities here, it's reading - restore past function